lib/yang_translator.c

```diff
@@ -171,56 +173,61 @@ struct yang_translator *yang_translator_load(const char *path)
 	RB_INSERT(yang_translators, &yang_translators, translator);
 
 	/* Initialize the translator libyang context. */
-	translator->ly_ctx = yang_ctx_new_setup(false);
+	translator->ly_ctx = yang_ctx_new_setup(false, false);
 	if (!translator->ly_ctx) {
 		flog_warn(EC_LIB_LIBYANG, "%s: ly_ctx_new() failed", __func__);
 		goto error;
 	}
 
 	/* Load modules and deviations. */
-	set = lyd_find_path(dnode, "./module");
-	assert(set);
-	for (size_t i = 0; i < set->number; i++) {
+	if (lyd_find_xpath(dnode, "./module", &set) != LY_SUCCESS)
+		assert(0); /* XXX libyang2: old ly1 code asserted success */
+
+	for (size_t i = 0; i < set->count; i++) {
 		const char *module_name;
 
 		tmodule =
 			XCALLOC(MTYPE_YANG_TRANSLATOR_MODULE, sizeof(*tmodule));
 
-		module_name = yang_dnode_get_string(set->set.d[i], "./name");
+		module_name = yang_dnode_get_string(set->dnodes[i], "./name");
 		tmodule->module = ly_ctx_load_module(translator->ly_ctx,
-						     module_name, NULL);
+						     module_name, NULL, NULL);
 		if (!tmodule->module) {
 			flog_warn(EC_LIB_YANG_TRANSLATOR_LOAD,
 				  "%s: failed to load module: %s", __func__,
 				  module_name);
-			ly_set_free(set);
+			ly_set_free(set, NULL);
 			goto error;
 		}
 
 		module_name =
-			yang_dnode_get_string(set->set.d[i], "./deviations");
-		tmodule->deviations = ly_ctx_load_module(translator->ly_ctx,
-							 module_name, NULL);
+			yang_dnode_get_string(set->dnodes[i], "./deviations");
+		tmodule->deviations = ly_ctx_load_module(
+			translator->ly_ctx, module_name, NULL, NULL);
 		if (!tmodule->deviations) {
 			flog_warn(EC_LIB_YANG_TRANSLATOR_LOAD,
 				  "%s: failed to load module: %s", __func__,
 				  module_name);
-			ly_set_free(set);
+			ly_set_free(set, NULL);
 			goto error;
 		}
-		lys_set_disabled(tmodule->deviations);
+
+		/*
+		 * XXX Previously in libyang1 tmodule->deviations was set disabled
+		 * here. This affected the counted nodes below. This disabling is
+		 * not implemented in libyang2 as compiled modules do not include
+		 * disabled nodes. Not sure how to fix this.
```

Comment:
I'm not familiar with this code but the comment sounds like something that should be fixed.

Reply:
well it's going to currently claim 100% coverage b/c it's not doing anything between before and after counts. :)

Reply:
I've fixed the code. Having trouble testing this though